### PR TITLE
test: remaining coverage of fy-input-popover comp

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at dfd5695</samp>

Improved unit testing for `fy-input-popover` component. Added tests for input access, saving, and validation. Covered different scenarios and edge cases.

### <samp>🤖 Generated by Copilot at dfd5695</samp>

> _To test `fy-input-popover`_
> _Some unit tests they did cover_
> _The input and save_
> _And how to behave_
> _When the input is not proper_

### <samp>🤖 Generated by Copilot at dfd5695</samp>

* Imported `getElementRef` function to access input element in tests ([link](https://github.com/fylein/fyle-mobile-app/pull/2412/files?diff=unified&w=0#diff-6008c7ebd66841dc1b0938b837903ad9f9a5091e60108f6b0f73aa73bab298d7L6-R6))
* Tested `saveValue` function for valid and invalid input values and popover dismissal ([link](https://github.com/fylein/fyle-mobile-app/pull/2412/files?diff=unified&w=0#diff-6008c7ebd66841dc1b0938b837903ad9f9a5091e60108f6b0f73aa73bab298d7L47-R69))
* Tested `onFocus`, `ngAfterViewInit` and `validateInput` functions for input focus, error message and validation logic ([link](https://github.com/fylein/fyle-mobile-app/pull/2412/files?diff=unified&w=0#diff-6008c7ebd66841dc1b0938b837903ad9f9a5091e60108f6b0f73aa73bab298d7R98-R150))

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI change

### CODE REVIEW

**Review by Dimple16 — APPROVED** [review]

File: src/app/shared/components/fy-input-popover/fy-input-popover.component.spec.ts [hunk 1]
```diff
@@ -44,15 +44,27 @@ describe('FyInputPopoverComponent', () => {
     expect(popoverController.dismiss).toHaveBeenCalledTimes(1);
   }));
 
-  it('saveValue(): should save input value', fakeAsync(() => {
-    popoverController.dismiss.and.returnValue(Promise.resolve(true));
-    component.inputValue = 'input';
-    fixture.detectChanges();
+  describe('saveValue():', () => {
+    beforeEach(() => {
+      popoverController.dismiss.and.returnValue(Promise.resolve(true));
```
Comment (OmkarJ13) [inline src/app/shared/components/fy-input-popover/fy-input-popover.component.spec.ts:49]: and.resolveTo?

File: src/app/shared/components/fy-input-popover/fy-input-popover.component.spec.ts [hunk 2]
```diff
@@ -44,15 +44,27 @@ describe('FyInputPopoverComponent', () => {
...
-    expect(popoverController.dismiss).toHaveBeenCalledOnceWith({ newValue: component.inputValue });
-  }));
+    it('should save input value', fakeAsync(() => {
+      fixture.detectChanges();
+
+      component.saveValue();
+      expect(popoverController.dismiss).toHaveBeenCalledOnceWith({ newValue: component.inputValue });
+    }));
+
+    it('should not call popoverController.dismiss if error length is greater than 0', () => {
```
Comment (OmkarJ13) [inline src/app/shared/components/fy-input-popover/fy-input-popover.component.spec.ts:60]: Let's rephrase this to something like: `should not save the input value and close the popover if the input value is invalid`

This reflects the behaviour more than the actual implementation details, and is easier to understand IMHO

File: src/app/shared/components/fy-input-popover/fy-input-popover.component.spec.ts [hunk 3]
```diff
@@ -82,4 +94,57 @@ describe('FyInputPopoverComponent', () => {
...
+    component.error = 'Please enter valid mobile number';
+    fixture.detectChanges();
+
+    component.onFocus();
+    expect(component.error).toEqual(null);
+  });
+
+  it('ngAfterViewInit(): should focus on input element', fakeAsync(() => {
+    const inputEl = getElementRef(fixture, 'input');
+    const focusSpy = spyOn(inputEl.nativeElement, 'focus');
```
Comment (OmkarJ13) [inline src/app/shared/components/fy-input-popover/fy-input-popover.component.spec.ts:108]: We don't need to store the spy ig, we can directly do `expect(input.nativeElement.focus).toHaveBeenCalledTimes(1)` after doing `spyOn`

**Review by OmkarJ13 — APPROVED** [review]
